pyccel/parser/syntactic.py

```diff
@@ -739,7 +739,12 @@ def fill_types(ls):
                 header = hdr_parse(stmts=txt)
                 if name in self.namespace.static_functions:
                     header = header.to_static()
-                headers += [header]
+                if all(header.dtypes != hd.dtypes for hd in headers):
+                    headers += [header]
+                else:
+                    errors.report(DUPLICATED_SIGNATURE, symbol=header,
+                        bounding_box=(stmt.lineno, stmt.col_offset),
+                        severity='warning')
```

Comment:
This code block appears 4 times in total. Is there no way to treat them all in the same place?

Reply:
We can keep the duplicated headers and decorators, until we gather them in `visit_functiondef` in semantics.

Reply:
That sounds better to me. It seems strange to not do them all at once

Reply:
So this will be handled in a separate PR? Do we have an open issue for it?

Reply:
I now raise the error only one time